### DISCUSSION TITLE
Building deb package on Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by ansluk** [edrlab/thorium-reader#2059] [General]
I wanted an epub reader with video/audio support on the raspberry pi 5, but knowing nothing of nodejs/npm/electron it wasn't immediately clear to me how to get thorium reader installed.
The steps I took to get it running on Raspberry Pi OS bookworm 64-bit are listed below in case someone else has problems building a deb.
```
sudo apt-get install npm
sudo apt-get install ruby
sudo gem install fpm
git clone https://github.com/edrlab/thorium-reader.git
cd thorium-reader/
```
Change architecture to arm64 and add the USE_SYSTEM_FPM environment variable to the package:linux line in the package.json file:

`"package:linux": "cross-env DEBUG=* CSC_IDENTITY_AUTO_DISCOVERY=false USE_SYSTEM_FPM=true electron-builder --linux --arm64 --publish never"`

USE_SYSTEM_FPM=true is necessary to not automatically download the wrong version (x86) of fpm.
(https://www.beekeeperstudio.io/blog/electron-apps-for-arm-and-raspberry-pi)

```
npm i --arch=arm64
npm run package:linux
sudo apt install ./release/EDRLab.ThoriumReader_2.4.0-alpha.1_arm64.deb
```

**Comment by danielweck** [comment]
Thorium's Docker builder for Linux also uses the `USE_SYSTEM_FPM` trick:

https://github.com/edrlab/thorium-reader/blob/7e60288f293a796ef7c59584d889b73f6db74a55/Dockerfile#L124C29-L124C46

https://github.com/edrlab/thorium-reader/blob/7e60288f293a796ef7c59584d889b73f6db74a55/Dockerfile#L46-L51

**Reply by danielweck** [reply]
...and as for ARM64 the Docker script automatically switches architecture based on the host platform (this was mainly created to generate Linux releases from MacOS Intel / AppleSilicon): https://github.com/edrlab/thorium-reader/blob/7e60288f293a796ef7c59584d889b73f6db74a55/docker.sh#L3-L8
